src/modules/billing/services/charge-processor-service/index.js

```diff
@@ -31,24 +31,26 @@ const createInvoiceLicence = (chargeVersion) => {
   });
 };
 
+/**
+ * Validate that all required models are present
+ * @param {ChargeVersionYear} chargeVersionYear
+ */
+const validateData = chargeVersionYear => {
+  validators.assertIsInstanceOf(chargeVersionYear, ChargeVersionYear);
+  validators.assertIsInstanceOf(chargeVersionYear.batch, Batch);
+  validators.assertIsInstanceOf(chargeVersionYear.chargeVersion, ChargeVersion);
+  validators.assertIsInstanceOf(chargeVersionYear.financialYear, FinancialYear);
+};
+
 /**
  * Creates the invoice data structure for a given charge version year
- * @param {Batch} batch
- * @param {FinancialYear} financialYear
- * @param {String} chargeVersionId
+ * @param {ChargeVersionYear} chargeVersionYear contains batch and chargeVersion
  * @return {Promise<Invoice>}
  */
-const processChargeVersionYear = async (batch, financialYear, chargeVersionId) => {
-  validators.assertIsInstanceOf(batch, Batch);
-  validators.assertIsInstanceOf(financialYear, FinancialYear);
-  validators.assertId(chargeVersionId);
-
-  // Load ChargeVersion service model
-  const chargeVersion = await chargeVersionService.getByChargeVersionId(chargeVersionId);
-  if (!chargeVersion) {
-    throw new NotFoundError(`Charge version ${chargeVersionId} not found`);
-  }
+const processChargeVersionYear = async (chargeVersionYear) => {
+  validateData(chargeVersionYear);
 
+  const { chargeVersion, financialYear } = chargeVersionYear;
   // Load all billing volumes relating to the charge elements and financial year
   // For supplementary these could have been generated by a previous TPT batch, not the current batch
   // There could also be multiple billing volumes per element - for summer and winter/all year TPT batches
```

Comment:
Could we skip loading the billingVolumes here if the `chargeVersionYear.transactionType` is `annual`?